tests/trainer/logging_/test_eval_loop_logging.py

```diff
@@ -924,3 +925,57 @@ def test_rich_print_results(inputs, expected):
     EvaluationLoop._print_results(*inputs, file=out)
     expected = expected[1:]  # remove the initial line break from the """ string
     assert out.getvalue() == expected.lstrip()
+
+
+@pytest.mark.parametrize("num_dataloaders", [1, 2])
+def test_eval_step_logging(tmpdir, num_dataloaders):
+    class CustomLogger(CSVLogger):
+        def __init__(self, *args, **kwargs):
+            super().__init__(*args, **kwargs)
+            self.logged_metrics = collections.defaultdict(lambda: collections.defaultdict(list))
+
+        def log_metrics(self, metrics, step):
+            for key in metrics:
+                self.logged_metrics[key]["logged_values"].append(metrics[key])
+                self.logged_metrics[key]["logged_steps"].append(step)
+
+    class CustomBoringModel(BoringModel):
+        def validation_step(self, batch, batch_idx, dataloader_idx=None):
+            self.log(f"val_log_{self.trainer.state.fn}", batch_idx, on_step=True, on_epoch=False)
+
+        def test_step(self, batch, batch_idx, dataloader_idx=None):
+            self.log("test_log", batch_idx, on_step=True, on_epoch=False)
+
+        def val_dataloader(self):
+            return [super().val_dataloader()] * num_dataloaders
+
+        def test_dataloader(self):
+            return [super().test_dataloader()] * num_dataloaders
+
+    logger = CustomLogger(tmpdir)
+    limit_batches = 4
+    max_epochs = 3
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        max_epochs=max_epochs,
+        limit_train_batches=1,
+        limit_val_batches=limit_batches,
+        limit_test_batches=limit_batches,
+        logger=logger,
+    )
+    model = CustomBoringModel()
+    model.validation_epoch_end = None
+    model.test_epoch_end = None
+
+    trainer.fit(model)
+    trainer.validate(model)
+    trainer.test(model)
+
+    for dl_idx in range(num_dataloaders):
+        suffix = f"/dataloader_idx_{dl_idx}" if num_dataloaders == 2 else ""
+        assert logger.logged_metrics[f"val_log_fit{suffix}"]["logged_values"] == list(range(limit_batches)) * max_epochs
+        assert logger.logged_metrics[f"val_log_fit{suffix}"]["logged_steps"] == list(range(limit_batches * max_epochs))
+        assert logger.logged_metrics[f"val_log_validate{suffix}"]["logged_values"] == list(range(limit_batches))
+        assert logger.logged_metrics[f"val_log_validate{suffix}"]["logged_values"] == list(range(limit_batches))
+        assert logger.logged_metrics[f"test_log{suffix}"]["logged_steps"] == list(range(limit_batches))
+        assert logger.logged_metrics[f"test_log{suffix}"]["logged_steps"] == list(range(limit_batches))
```

Comment:
Some duplication here:
```suggestion
        assert logger.logged_metrics[f"val_log_validate{suffix}"]["logged_values"] == list(range(limit_batches))
        assert logger.logged_metrics[f"test_log{suffix}"]["logged_steps"] == list(range(limit_batches))
```

Reply:
good catch!!